nomad/state/state_store.go

```diff
@@ -206,10 +206,16 @@ func (s *StateStore) AbandonCh() <-chan struct{} {
 // Abandon is used to signal that the given state store has been abandoned.
 // Calling this more than one time will panic.
 func (s *StateStore) Abandon() {
-	s.stopEventPublisher()
+	s.StopEventPublisher()
 	close(s.abandonCh)
 }
 
+// StopStopEventPublisher calls the cancel func for the state stores event
+// publisher. It should be called during server shutdown.
+func (s *StateStore) StopEventPublisher() {
+	s.stopEventPublisher()
+}
+
```

Comment:
Why not just call Abandon in tests?

Reply:
StopEventPublisher will be used during server shutdown, the state store tests can call Abandon(), when I had Abandon() being called during fsm/server shutdown it caused weird leadership flapping https://app.circleci.com/pipelines/github/hashicorp/nomad/11813/workflows/611f362f-009a-497f-a0e2-60cc25977fb6/jobs/100908